tests/link/jax/test_sparse.py

```diff
@@ -2,13 +2,51 @@
 import pytest
 import scipy.sparse
 
+
+jax = pytest.importorskip("jax")
+from jax.experimental.sparse import BCOO
+
 import pytensor.sparse as ps
 import pytensor.tensor as pt
 from pytensor import function
-from pytensor.graph import FunctionGraph
+from pytensor.graph import Constant, FunctionGraph
+from pytensor.tensor.type import DenseTensorType
 from tests.link.jax.test_basic import compare_jax_and_py
 
 
+def assert_bcoo_arrays_allclose(a1, a2):
+    assert isinstance(a1, BCOO)
+    assert isinstance(a1, BCOO)
+    np.testing.assert_allclose(a1.todense(), a2.todense())
+
+
+@pytest.mark.parametrize("sparse_type", ("csc", "csr"))
+def test_sparse_io(sparse_type):
+    """Test explicit (non-shared) input and output sparse types in JAX."""
+    sparse_mat = ps.matrix(format=sparse_type, name="csc", dtype="float32")
```

Comment:
Why not `floatX` here (and below)

Reply:
JAX doesn't like float64, so this probably avoids annoying warnings from them

Reply:
In general the floatX thing from the times of Theano is a very lazy way of testing one specific and back then popular mode "float32/" (or "FAST_COMPILE") doesn't accidentally break. This is a bit silly, because there's a thousand other configs that one could argue are also important to test and we do not (e.g., no optimization at all). 

What I am trying to say is that the float32/float64 tests are not a reasonable way to actually check our code works with custom configurations. For legacy we keep doing it in old tests but I don't really try to introduce them all the time in new tests unless I foresee a reason why float32/float64 flexibility should be tested specifically.